lib/rule_runner.rb

```diff
@@ -8,17 +8,23 @@ def initialize(rule:, callback:, shopify_identifier:)
   def perform
     rule_event.add_detail(:info, "Event received")
 
-    filters_valid = true
-    rule.filters.order(:id).each.with_index do |filter, index|
-      if FilterValidator.new(filter).valid?(callback)
-        rule_event.add_detail(:info, "Filter ##{index + 1}: Met")
-      else
-        rule_event.add_detail(:info, "Filter ##{index + 1}: Unmet")
-        filters_valid = false
+    filters_results = rule.filters.order(:id).map.with_index do |filter, index|
+      begin
+        if FilterValidator.new(filter).valid?(callback)
+          rule_event.add_detail(:info, "Filter ##{index + 1}: Met")
+          true
+        else
+          rule_event.add_detail(:info, "Filter ##{index + 1}: Unmet")
+          false
+        end
+      rescue StandardError => e
+        rule_event.add_detail(:error, "Filter ##{index + 1}: There was an error applying this filter")
+        Rails.logger.info("User error: #{e.message}")
+        false
```

Comment:
I disagree with rescuing and silencing server exceptions. 

This will lead an error that only Triggerify devs can solve to be silenced from Bugsnag and to be shown to the user instead.

Is there a specific case you are trying to handle that could be coded differently?